version: '3.9'

services:
  external_api_gateway:
    build: ./EXTERNAL_API_GATEWAY
    environment:
      - HOST=${EXTERNAL_API_GATEWAY_HOST}
      - PORT=${EXTERNAL_API_GATEWAY_APP_PORT}
    ports:
      - ${EXTERNAL_API_GATEWAY_PORT}
    volumes:
      - ./EXTERNAL_API_GATEWAY:/app
    restart: on-failure  

  service_math_solve:
    build: ./SERVICE_MATH_SOLVE 
    environment:
      - HOST=${MATH_SOLVE_HOST}
      - PORT=${MATH_SOLVE_APP_PORT}
    ports:
      - ${MATH_SOLVE_PORT}
    volumes:
      - ./SERVICE_MATH_SOLVE:/code
    restart: on-failure

  redis:
    image: redis:alpine
    ports:
      - "6379:6379"
    restart: on-failure

  postgres:
    image: postgres:14
    environment:
      - POSTGRES_USER=user
      - POSTGRES_PASSWORD=pass
    volumes:
      - postgres_data:/var/lib/postgresql/data
    restart: on-failure

volumes:
  postgres_data:
    driver: local